### PR TITLE
[MLIR][OpenMP][Docs] NFC: Document compound constructs representation

### PR DESCRIPTION
**Pull request by skatrak**
This patch documents the MLIR representation of OpenMP compound constructs discussed in [this](https://discourse.llvm.org/t/rfc-representing-combined-composite-constructs-in-the-openmp-dialect/76986) and [this](https://discourse.llvm.org/t/rfc-disambiguation-between-loop-and-block-associated-omp-parallelop/79972) RFC.

### CODE REVIEW

**Comment by skatrak** [comment]
PR Stack:
- #107232
- #107233
- #107234
- #107235
- #107236

File: mlir/docs/Dialects/OpenMPDialect/_index.md [hunk 1]
```diff
@@ -287,3 +287,117 @@ been implemented, but it is closely linked to the `omp.canonical_loop` work.
...
+`distribute parallel {do,for}` composite construct. The presence of a
+block-associated `parallel` leaf construct would introduce many problems if it
+was allowed to work as a loop wrapper. In this case, the "hoisted `omp.parallel`
+representation" is used instead. This consists in making `omp.parallel` the
+parent operation, with a nested `omp.loop_nest` wrapped by `omp.distribute` and
+`omp.wsloop` (and `omp.simd`, in the `distribute parallel {do,for} simd` case).
+
+This approach works because `parallel` is a parallelism-generating construct,
+whereas `distribute` is a worksharing construct impacting the higher level
+`teams`, making the ordering between these constructs not cause semantic
```
Comment (mjklemm) [inline mlir/docs/Dialects/OpenMPDialect/_index.md:384]:
```suggestion
`teams` construct, making the ordering between these constructs not cause semantic
```

**Review by mjklemm — APPROVED** [review]

**Review by tblah — APPROVED** [review]
LGTM, thank you for writing all of these

File: mlir/docs/Dialects/OpenMPDialect/_index.md (hunk 1 above)
Comment (skatrak) [reply mlir/docs/Dialects/OpenMPDialect/_index.md]: Thank you for noticing, done.

**Pull request merged**